hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -424,15 +424,15 @@ namespace hpx { namespace threads { namespace detail
         scheduling_counters& counters, scheduling_callbacks& params)
     {
         std::atomic<hpx::state>& this_state = scheduler.get_state(num_thread);
-
+#ifdef HPX_HAVE_ITTNOTIFY
```

Comment:
@biddisco I don't think this is necessary as all the types in `util::itt` are defined as being empty if `HPX_HAVE_ITTNOTIFY` was not defined (see: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/itt_notify.hpp#L619-L731).

Reply:
Putting the #ifdef around that block does prevent the allocation of an empty object though.

Reply:
That gets optimized away, for sure. Even if not, it's just one byte on the stack.

Reply:
The call to 
```
        util::itt::domain domain = hpx::get_thread_itt_domain();
```
allocates an itt::domain
```
        if (nullptr == d.get())
            d.reset(new util::itt::domain(get_thread_name().c_str()));
```
and this is a full malloc. Not a single byte on the stack. I don't use ITTNOTIFY and I don't see why I should be allocating any of this. I think the #ifdef should stay.

Reply:
It does not allocate anything if `ITTNOTIFY` is disabled, see here: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/itt_notify.hpp#L619-L731. At least that's the intention.

Reply:
I think you're both right.

On current master a struct is always allocated on the heap no matter if ITT is enabled or not. With ITT off the struct is empty and the use in `scheduling_loop` is ifdefed out, so it might get optimized away.

If I get my RAII wrapper done it would indeed be an empty struct on the stack when ITT is off. Chances of this getting optimized out are higher, but I don't know if it would happen.